### PR TITLE
Wes/integ 417 motherduck blueprints

### PR DESCRIPTION
**Pull request by wrp801**

### CODE REVIEW

File: shipyard_blueprints/motherduck/shipyard_motherduck/cli/upload.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+                logger.error(f"No file matches found for pattern {file_name}")
+                sys.exit(Database.EXIT_CODE_FILE_NOT_FOUND)
+            logger.info(f"Found {n_matches} file matches, preparing to upload...")
+            for file in file_matches:
+                logger.debug(f"Uploading {file}")
+                client.upload(table_name, file, insert_method=insert_method)
+                if insert_method == "replace":
+                    insert_method = "append"
+            logger.info("Successfully uploaded all files")
+        else:
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/motherduck/shipyard_motherduck/cli/upload.py:65]: Can we make this explicit

Comment (johnathan-rodriguez) [inline shipyard_blueprints/motherduck/shipyard_motherduck/cli/upload.py:65]: Can we make this explicit

File: shipyard_blueprints/motherduck/shipyard_motherduck/errors.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from shipyard_templates import ExitCodeException
+
+EXIT_CODE_TEMP_TABLE_CREATION_ERROR = 101
+EXIT_CODE_TEMP_TABLE_INSERT_ERROR = 102
+EXIT_CODE_TEMP_TABLE_DROP_ERROR = 103
+
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/motherduck/shipyard_motherduck/errors.py:6]: can we name this file exceptions to be more consistent with the other blueprints.

File: shipyard_blueprints/motherduck/shipyard_motherduck/motherduck.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+from typing import Optional
+
+logger = ShipyardLogger().get_logger()
+
+
+class MotherDuckClient(Database):
+    def __init__(self, token: str, database: Optional[str] = None):
+        self.token = token
+        self.database = database
+        if database:
```
Comment (johnathan-rodriguez) [inline shipyard_blueprints/motherduck/shipyard_motherduck/motherduck.py:23]: Just worth noting: In the cli you're explicitly passing None where empty string but that logic seems to already be handled here so it is not needed in the scripts

**Review by johnathan-rodriguez — CHANGES_REQUESTED** [review]
Very minor changes to make the blueprint more consistent with existing blueprints

File: shipyard_blueprints/motherduck/shipyard_motherduck/cli/upload.py (hunk 1 above)
Comment (wrp801) [reply shipyard_blueprints/motherduck/shipyard_motherduck/cli/upload.py:65]: explicit in what sense?

Comment (johnathan-rodriguez) [reply shipyard_blueprints/motherduck/shipyard_motherduck/cli/upload.py:65]: `args.match_type=='exact_match`

**Review by johnathan-rodriguez — APPROVED** [review]

**Pull request merged**